### DISCUSSION TITLE
concatenate commands for compilation

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#209] [Q&A] [answered]
I’m testing Textadept as a replacement (or simply an alternative) to Geany.

I don’t code and I’m stuck with Markdown compilation (which is pandoc plus ConTeXt): 

``` lua
textadept.run.compile_commands.markdown = 'pandoc -o "%e.xml" "%f" && context --environment="%e".tex "%e".xml'
```

Textadept doesn’t accept two commands, but how can I get both when I triigger the complation?

Many thanks for your help and your excellent work.

**Comment by ousia** [comment]
Sorry for trying this again, but is there no way in which I can concatenate commands such as in `&&` for `textadept.run.compile_commands`?

**Reply by rgieseke** [reply]
Not sure if there is a way in which Textadept prevents this, but maybe you could call a bash (or other) script which runs your commands.

**Comment by orbitalquark** [comment, marked as answer]
Textadept does not support this. You could try something like:

  textadept.run.compile_commands.markdown = function()
    local xml = buffer.filename:gsub('%.md$', '.xml')
    local status = os.spawn(string.format('pandoc -o "%s" "%s"', xml, buffer.filename'), ui.print, ui.print):wait()
    if status ~= 0 then return nil end
    return 'context --environment="%e.tex" "%e.xml"'
  end

The command will manually run your pandoc command and then return the context command for Textadept to run using its method.

**Reply by ousia** [reply]
Many thanks for your code, @orbitalquark.

I didn’t know it was so complex.

Just out of curiosity, does the lack of support comes from Scintilla or is this something totally unrelated to it?

Long time ago, I wanted to do the same with Notepad++ and I think I had the same issue (or a very similar one).

Many thanks for your help.

**Reply by ousia** [reply]
@orbitalquark,

I have just discovered that this does the job:

``` lua
textadept.run.compile_commands.markdown = function()
  return 'pandoc -o "%e.xml" "%f" && context --environment="%e".tex "%e".xml'
end
```

The other option doesn’t work the Windows setup I use (this is why I discovered that the function above worked).

Is there something wrong with the proposed function?

Many thanks for your help.

**Comment by orbitalquark** [comment]
Totally unrelated. The compile and run code passes commands to `os.spawn()`, which can only run a single command at a time.

**Reply by ousia** [reply]
Many thanks for your reply, I see that this is totally different.

I’m afraid that Textadept complains about the `os.spawn` in your sample:

```lua
lua: /home/ousia/.textadept/init.lua:12: unfinished string near ''), ui.print, ui.print):wait()'
```

I don’t know what is wrong here.

**Comment by orbitalquark** [comment]
Try ` local status = os.spawn(string.format('pandoc -o "%s" "%s"', xml, buffer.filename), ui.print, ui.print):wait()`

Looks like I made a typo.

**Reply by ousia** [reply]
I see now the extra `'` removed.

It works perfectly fine now.

Many thanks for your help.

**Comment by orbitalquark** [comment]
Interesting. Thanks for following up. I don't recall actually testing the function. It was more of an idea to communicate.

**Reply by ousia** [reply]
As I discovered this, I thought there might be something wrong (or a catch 😅).
